src/dash/parser/objectiron.js

```diff
@@ -32,37 +32,54 @@ import FactoryMaker from '../../core/FactoryMaker.js';
 
 function ObjectIron(mappers) {
 
-    function mergeValues(parentItem, childItem) {
-        for (let name in parentItem) {
-            if (!childItem.hasOwnProperty(name)) {
-                childItem[name] = parentItem[name];
+    function _conditionallyMapProperty(exception, parentName, parentIsArray, parentEl, child, mergeFlag) {
+        let allowMapping = true;
+
+        // check, if element matches an exception
+        if (exception) {
+            for (const [key, values] of Object.entries(exception)) {
+                let attr = parentEl[key];
+                if (values.some(v => attr.match(v))) {
+                    allowMapping = false;
+                }
+            }
+        }
+
+        // apply mapping
+        if (allowMapping) {
+            if (child[parentName]) {
+                // property already exists
+                // check to see if we should merge
+                if (mergeFlag) {
+                    if (parentIsArray) {
+                        child[parentName].push(parentEl);
+                    }
+                }
+            } else {
+                // just add the property
+                if (parentIsArray) {
+                    child[parentName] = [parentEl];
+                } else {
+                    child[parentName] = parentEl;
+                }
             }
         }
+
     }
 
-    function mapProperties(properties, parent, child) {
+    function mapProperties(properties, exceptions, parent, child) {
         for (let i = 0, len = properties.length; i < len; ++i) {
             const property = properties[i];
 
             if (parent[property.name]) {
-                if (child[property.name]) {
-                    // check to see if we should merge
-                    if (property.merge) {
-                        const parentValue = parent[property.name];
-                        const childValue = child[property.name];
-
-                        // complex objects; merge properties
-                        if (typeof parentValue === 'object' && typeof childValue === 'object') {
-                            mergeValues(parentValue, childValue);
-                        }
-                        // simple objects; merge them together
-                        else {
-                            child[property.name] = parentValue + childValue;
-                        }
-                    }
+                const propertyParentElementArray = parent[property.name];
```

Comment:
The name `propertyParentElementArray` might be a bit confusing as we are checking afterwards if this is really an array or not. I suggest to rename it to `propertyParentElement`